### PR TITLE
Added za() function to identifiers

### PR DESCRIPTION
**Pull request by whaeck**
Added a za() function to particle identifiers

### CODE REVIEW

File: src/elementary/FundamentalParticleID.hpp [hunk 1]
```diff
@@ -40,10 +40,15 @@ namespace elementary {
...
+    /**
+     *  @brief return the fundamental particle za value
+     */
+    int za() const noexcept { return this->hash_; }
+
     /**
      *  @brief return the fundamental particle hash value
      */
-    int hash() const noexcept { return this->hash_; }
+    int hash() const noexcept { return this->za(); }
```
Comment (jlconlin) [inline src/elementary/FundamentalParticleID.hpp:51]: Why have two names for this? Why not just have `za` and get rid of `hash`?

File: src/elementary/FundamentalParticleID/src/register.hpp [hunk 2]
```diff
@@ -6,8 +6,8 @@ FundamentalParticleID::particle_dictionary{
 
   { 0, FundamentalParticleID::Entry{ 0, "photon", "photon",
                                      { "g", "gamma", "x-ray" } } },
-  { 10, FundamentalParticleID::Entry{ 10, "n", "neutron" } },
-  { 10010, FundamentalParticleID::Entry{ 10010, "p", "proton" } }
+  { 1, FundamentalParticleID::Entry{ 1, "n", "neutron" } },
```
Comment (jlconlin) [inline src/elementary/FundamentalParticleID/src/register.hpp:9]: I like this a lot better.

File: src/elementary/NuclideID.hpp [hunk 3]
```diff
@@ -86,12 +86,20 @@ namespace elementary {
...
+     */
+    int za() const noexcept {
+
+      return this->isotope_.za();
+    }
+
     /**
      *  @brief return the nuclide hash value
      */
     int hash() const noexcept {
```
Comment (jlconlin) [inline src/elementary/NuclideID.hpp:100]: Again, why have `hash`?

File: src/elementary/FundamentalParticleID.hpp (hunk 1 above)
Comment (whaeck) [reply src/elementary/FundamentalParticleID.hpp:51]: I wanted the same interface for all identifiers. For the fundamental particle, the hash is the za. For the nucleus and the nuclide it is za * 100 + level. The particle identifier also makes a hash which is again different, depending on the underlying type.

Comment (whaeck) [reply src/elementary/FundamentalParticleID.hpp:51]: Maybe at some point we'll change the hash, and by already providing the function to distinguish between za and hash, I can avoid changing too much later.

**Pull request merged**